Classes/common/touchdb/TDMisc.m

```diff
@@ -258,3 +258,41 @@ BOOL TDMayBeTransientError(NSError* error)
     [urlStr appendString:toAppend];
     return [NSURL URLWithString:urlStr];
 }
+
+NSString* TDCleanURLtoString(NSURL* url)
+{
+    if (!(url.scheme && url.host)) {
+        //first, "validate" the URL. We do this to ensure, at a very basic level, that the URL was
+        //not initialized improperly (via a typo, for example, in URLwithString)
+        //the NSURL should have a scheme and host. Otherwise, it won't successfully parse
+        //out the user/password. As a result, we return the following error string.
+        //Obviously, however, we cannot protect against all kinds of typos that leak credentials.
+        return @"NSURL did not have both a scheme and host. Suppressing string value for security.";
```

Comment:
I don't think this is a very complete check, and so more likely to cause harm than good. I'd remove it.